pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -331,51 +338,114 @@ func updateFromSecret(coreClient controllerclient.Client, scope *MachineScope) e
 		return err
 	}
 
-	subscriptionID, ok := secret.Data[AzureCredsSubscriptionIDKey]
+	subscriptionIDBytes, ok := secret.Data[AzureCredsSubscriptionIDKey]
 	if !ok {
 		return fmt.Errorf("Azure subscription id %v did not contain key %v",
 			secretType.String(), AzureCredsSubscriptionIDKey)
 	}
-	clientID, ok := secret.Data[AzureCredsClientIDKey]
+	clientIDBytes, ok := secret.Data[AzureCredsClientIDKey]
```

Comment:
A lot of the code in this function is duplicated many times, I wonder if extracting it to a helper that first tries the secret and then tries the env var would make this cleaner, WDYT?